Better readme

tobbe-j opened the pull request:
Better readme to help new users or developers.

Ideas:
- quick jump on top -- added ToC
- quick jump to top -- added go-to-top link 
- quick jumps to demo (video) -- added link to demo video
- add credit to github users -- add links to all github users of people involved
- add link to landing page -- added link to landing page


## Review Checklist

- [x] All changes are described in the PR description
- [x] New feature and breaking changes have documentation
- [x] New endpoints have API tests
- [x] New frontend features have Enzyme tests
- [x] No unnecessary or debugging code
- [x] Questionable code is clearly marked with comments
- [x] Manually tested with latest Firefox, Safari and Chrome

tobbe-j committed:
Added new headers and badges

tobbe-j committed:
Update readme

Added contributing section with how to run tests

tobbe-j committed:
Update readme

Added credits, license and a start revision for the introduction
motivation part.

tobbe-j commented:
![](https://github.trello.services/images/mini-trello-icon.png) [[task] [misc] Better github readme, prep for open source Tobias](https://trello.com/c/eJn5hlmv/171-task-misc-better-github-readme-prep-for-open-source-tobias)

tobbe-j committed:
Added images to readme

tobbe-j committed:
Added feature list to readme

HenrikHelenius commented on README.md:
```diff
@@ -37,8 +55,92 @@ docker-compose up --build
+[jest]: https://jestjs.io/
+[enzyme]: https://github.com/airbnb/enzyme
+[mocha]: https://mochajs.org/
+[aalto]: https://aalto.fi
```
https://www.aalto.fi/department-of-computer-science

HenrikHelenius commented on README.md:
```diff
@@ -37,8 +55,92 @@ docker-compose up --build
+
+## Credits
+
+This project was started during the course CS-C2130 Software Project at [Aalto University][aalto] for the company [Sievo][sievo].
```
Tag github users

HenrikHelenius replied:
The project was supported by Sievo

tobbe-j committed:
update readme

Made logo smaller

tobbe-j committed:
Update readme

Added table of contents

tobbe-j committed:
Swapped gif for static image

tobbe-j committed:
Added new image

tobbe-j committed:
update readme

Moved images to own section to make readme simpler

tobbe-j committed:
update readme

Fixed screenshot image markup

tobbe-j committed:
update readme

Added github name tags of all devs to credits

tobbe-j committed:
update readme

Fixed user links to devs by adding url links

tobbe-j committed:
update readme

Specified urls better

tobbe-j committed:
update readme

Added links to demo video and landing page

HenrikHelenius commented on README.md:
```diff
@@ -1,8 +1,46 @@
-# Feedbacker-Forum
+<img src="docs/images/feedbacker-logo.png" height="50%" width="50%">
```
is margin possible? add negative space under

HenrikHelenius committed:
Update README.md

HenrikHelenius commented on README.md:
```diff
@@ -37,8 +77,114 @@ docker-compose up --build
+## Screenshots
+
+#### Demo
+Demo video [Feedbacker-Forum](https://www.youtube.com/watch?v=bkxmN-j-ae0)
```
Fake video embed like this: https://stackoverflow.com/questions/4279611/how-to-embed-a-video-into-github-readme-md

HenrikHelenius commented on README.md:
```diff
@@ -1,8 +1,46 @@
+The developer can then create surveys, recieve answers and discuss the live UI with peers.
+Feedbacker-Forum also supports creating "feedback-able" links for existing external websites.
+
+You can request access to try out Feedbacker-forum from [Feedbacker.site](https://feedbacker.site/about/)
```
Remove this, just put it like bla bla landing page: /about etc

HenrikHelenius commented on README.md:
```diff
@@ -37,8 +77,114 @@ docker-compose up --build
+
+This project was started during the course CS-C2130 Software Project at [Aalto University][aalto].
+The project was supported by [Sievo][sievo].
+The project was created by
```
created by:

bqqbarbhg commented on README.md:
```diff
@@ -37,8 +77,114 @@ docker-compose up --build
+This project was started during the course CS-C2130 Software Project at [Aalto University][aalto].
+The project was supported by [Sievo][sievo].
+The project was created by
+- [@bqqbarbhg](https://github.com/bqqbarbhg)
```
Why namedropping?

tobbe-j replied:
Considered unnecessary, removed

tobbe-j committed:
update readme

Changed video link to preview image

tobbe-j committed:
update readme

removed namedropping, centered logo and added whitespace under logo

tobbe-j committed:
Merge branch 'develop' into better-readme

joonalillfors commented on README.md:
```diff
@@ -1,8 +1,49 @@
+Feedbacker-Forum is a modern, open-source automated deployment pipeline solution for developers to help gather feedback easily. In other words the developer can trigger a deployment process that ships a version of the current code for the product and gets a simple “feedback-able“ link in return.
+
+ This link can be shared to peers that the developer craves feedback from.
+The developer can then create surveys, recieve answers and discuss the live UI with peers.
```
recieve -> receive

tobbe-j committed:
updated license

Updated year in license

joonalillfors commented on README.md:
```diff
@@ -1,8 +1,49 @@
+
+ This link can be shared to peers that the developer craves feedback from.
+The developer can then create surveys, recieve answers and discuss the live UI with peers.
+Feedbacker-Forum also supports creating "feedback-able" links for existing external websites [/about](https://feedbacker.site/about/).
```
what is the meaning of this /about?

joonalillfors replied:
it doesn't quite flow with the sentence

joonalillfors commented on README.md:
```diff
@@ -1,8 +1,49 @@
 
 ## Setting up
 
+The project consist of a frontend, backend, proxy and database. The different parts can either be setup manually, or run together using containers.
```
The project -> the application?

joonalillfors commented on README.md:
```diff
@@ -1,8 +1,49 @@
+### Prerequisites
+
 The project is built using [Node.js][node], currently actively tested with version 10.13.
+The project also requires [go][golang] version 1.11 for running the proxy.
```
project -> application?

joonalillfors commented on README.md:
```diff
@@ -37,8 +80,105 @@ docker-compose up --build
+
+#### Dashboard
+
+The dashboard view is for developers to add, remove and manage instances.
```
add -> create?

joonalillfors commented on README.md:
```diff
@@ -37,8 +80,105 @@ docker-compose up --build
+#### Dashboard
+
+The dashboard view is for developers to add, remove and manage instances.
+After creating an instance the developer recieves a simple link that can be shared, so that people easily can view the instance and give feedback.
```
recieves -> receives

joonalillfors replied:
people can easily view

tobbe-j committed:
Merge branch 'better-readme' of https://github.com/ktk-2005/Feedbacker-Forum into better-readme

joonalillfors commented on README.md:
```diff
@@ -37,8 +80,105 @@ docker-compose up --build
+#### Features
+
+A brief list of supported features
+* Automatically deploy your app from git, boh public repos and private Github repos supported
```
boh -> both, repos or repositorys?

joonalillfors commented on README.md:
```diff
@@ -37,8 +80,105 @@ docker-compose up --build
+A brief list of supported features
+* Automatically deploy your app from git, boh public repos and private Github repos supported
+* Alternatively use an already live site, simply submit the url
+* Easily discuss and cultivate ideas with theaded discussion, no blocking signup required
```
threaded discussion

joonalillfors commented on README.md:
```diff
@@ -37,8 +80,105 @@ docker-compose up --build
+* Easily discuss and cultivate ideas with theaded discussion, no blocking signup required
+* Tag specific elements to link your comment to a specific part of the UI
+* Comment anonymously on instances where you don't want to share your identity
+* Specify the feedback recieved by creating surveys with normal or multiple choice questions
```
received

joonalillfors replied:
free form or multiple choice questions?

joonalillfors commented on README.md:
```diff
@@ -37,8 +80,105 @@ docker-compose up --build
+* Comment anonymously on instances where you don't want to share your identity
+* Specify the feedback recieved by creating surveys with normal or multiple choice questions
+* Protect sensitive deployments with a passphrase
+* Boost productivity by integrating Feedbacker-Forum to slack with a SlackBot
```
Slack

joonalillfors commented on README.md:
```diff
@@ -37,8 +80,105 @@ docker-compose up --build
+
+## Contributing
+
+The project is still in a quite early development stage and is actively being developed.
```
project -> application?

joonalillfors commented on README.md:
```diff
@@ -37,8 +80,105 @@ docker-compose up --build
+## Contributing
+
+The project is still in a quite early development stage and is actively being developed.
+Changes or additions can be proposed by creating issues or pull requests on github. Running a local development version can be setup as seen in [Setting Up](#Setting-up).
```
as seen in -> as shown in?

tobbe-j committed:
Merge branch 'develop' into better-readme

joonalillfors commented on README.md:
```diff
@@ -37,8 +80,105 @@ docker-compose up --build
+
+### Running tests
+
+The project contains has style checking using [eslint][eslint] with a slightly modified [airbnb-config][airbnb-config].
```
the project contains has

tobbe-j committed:
updated readme

mutliple word changes
- spelling corrections
- project -> application
- minor fixes

tobbe-j committed:
update readme

Moved link to feedbacker.site to other location

tobbe-j committed:
Merge branch 'better-readme' of https://github.com/ktk-2005/Feedbacker-Forum into better-readme

tobbe-j committed:
changed wording about surveys

sofiatengvall committed:
Update README.md

HenrikHelenius committed:
Added roadmap

HenrikHelenius committed:
Roadmap?

HenrikHelenius committed:
Update README.md

sofiatengvall committed:
small typo fixes

quvide commented on README.md:
```diff
@@ -1,8 +1,47 @@
+The application also requires [go][golang] version 1.11 for running the proxy.
+Instances are run in containers with [docker][docker].
+
+If the application is run with containers [docker-compose][docker-compose] is also required for orchestrating the containers.
```
What does this mean? Docker-compose is not required when not running the app backend itself outside of docker

HenrikHelenius committed:
Update README.md

tobbe-j committed:
Removed unclear sentence

tobbe-j committed:
Merge branch 'develop' into better-readme

HenrikHelenius reviewed (approved):
🎃🎃🎃🎃🎃🎃🎃🎃🎃
